hoomd/pytest/test_box_resize.py

```diff
@@ -0,0 +1,128 @@
+from math import isclose
+import numpy as np
+import numpy.testing as npt
+import pytest
+import hoomd
+
+
+@pytest.fixture(scope="module")
+def fractional_coordinates(n=10):
+    """
+    TODO: Does `numpy_random_seed()` in conftest.py run for this function?
+    Args:
+        n: number of particles
+
+    Returns: absolute fractional coordinates
+
+    """
+    return np.random.uniform(-0.25, 0.25, size=(n, 3))
+
+
+@pytest.fixture(scope="module")
+def sys1(fractional_coordinates):
+    """Initial system box size and particle positions.
+    Args:
+        fractional_coordinates: Array of fractional coordinates
+
+    Returns: hoomd box object and points for the initial system
+
+    """
+    hoomd_box = hoomd.Box.from_box([1., 2., 3., 1., 2., 3.])
+    points = fractional_coordinates @ hoomd_box.matrix.T
+    return hoomd_box, points
+
+
+_box2 = [[50, 2., 3., 1., 2., 3.],  # Only change Lx
+         [1., 2., 3., 0.9, 2., 3.],   # Only change xy
+         [50., 2., 3., 0.9, 2., 3.],   # Change Lx and xy
+         [100., 200., 600., 0.9, 5., 7.]]  # Change all
+
+
+@pytest.fixture(scope='module', params=_box2)
+def sys2(request, fractional_coordinates):
+    hoomd_box = hoomd.Box.from_box(request.param)
+    points = fractional_coordinates @ hoomd_box.matrix.T
+    return hoomd_box, points
+
+
+@pytest.fixture(scope='function')
+def get_snapshot(sys1, device):
+    def make_shapshot():
+        box1, points1 = sys1
+        s = hoomd.Snapshot()
+        s.configuration.box = box1
+        s.particles.N = points1.shape[0]
+        s.particles.typeid[:] = [0] * points1.shape[0]
+        s.particles.types = ['A']
+        s.particles.position[:] = points1
+        return s
+    return make_shapshot
+
+
+_t_start = 2
+_t_a = 1
+_t_b = 1
+_t_ab = 2
+_t_ba = 1
+_variants = [
+    hoomd.variant.Power(0., 1., 0.1, _t_start, _t_start*2),
+    hoomd.variant.Ramp(0., 1., _t_start, _t_start*2),
+    hoomd.variant.Cycle(0., 1., _t_start, _t_a, _t_ab, _t_b, _t_ba)
+             ]
+
+
+@pytest.fixture(scope='function', params=_variants)
+def variant(request):
+    return request.param
+
+
+_scale_particles = [True,False]
+
+
+@pytest.fixture(scope='function', params=_scale_particles)
+def scale_particles(request):
+    return request.param
+
+
```

Comment:
For this fixture we could just use `pytest.mark.parameterize` since it is only used for this function and only for the simple values `True` and `False`.